src/System.Collections.Immutable/src/Properties/SecurityTransparent.cs

```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Security;
+
+[assembly: SecurityTransparent]
```

Comment:
The secannotate report is clean:

> Annotating 'System.Collections.Immutable'.
Beginning pass 1.
Pass complete, 0 new annotation(s) generated.

If I was more comfortable with the build process and knew it would work either cross platform or could constrain it to only happen on Windows, I could wire up secannotate to run as part of the build or test process. But as I think this library is unlikely to make any calls to Critical code, it's probably overkill to guard against introducing such unless the cost is low.